post/2023/rsi-bp1/index.qmd

```diff
@@ -1,10 +1,224 @@
+```
+
+`get_stac_data()` and its child functions return the path to downloaded image. 
+As a default, they create the composite image out of median values of all avaiable images for specified time span. 
+We can specify other composite functions, like `mean`, `sum`, `min` or `max`. 
+We can also not use composite function at all (`composite_function = NULL`), and obtain all available images for specified time span. 
+Another default argument, passed in for `mask_band`, automatically masks clouds with NAs.
+
+``` {r downloading sentinel2 for sa}
+sa_sentinel2 = get_sentinel2_imagery(
+    san_antonio,
+    start_date = "2023-09-01",
+    end_date = "2023-10-31",
+    output_filename = tempfile(fileext = ".tif")
+)
+```
+
+In this example, we download Sentinel 2 images for September and October 2023, mask out clouds, and create a composite image.
+
+Let's also obtain the DEM of our area.
+
+``` {r downloading dem for sa}
+sa_dem = get_dem(
+    san_antonio,
+    output_filename = tempfile(fileext = ".tif")
+)
+```
+
+After downloading the images, we can load them into `SpatRaster` objects using [`rast()`](https://rdrr.io/cran/terra/man/rast.html).
+**rsi** automatically assigns proper band names to it, which follow Awesome Spectral Indices standard.
+
+``` {r loading imagery as SpatRaster}
+library(terra)
+sa_landsat_rast = terra::rast(sa_landsat)
+sa_sentinel2_rast = terra::rast(sa_sentinel2)
+sa_dem_rast = terra::rast(sa_dem)
+sa_landsat_rast
+```
+
+We can plot the rasters together to compare Landsat and Sentinel images and visualize DEM.
+
+``` {r plotting rasters}
+par(mfrow = c(1, 3))
+terra::plotRGB(sa_landsat_rast, r = 4, g = 3, b = 2, stretch = "lin")
+terra::plotRGB(sa_sentinel2_rast, r = 4, g = 3, b = 2, stretch = "lin")
+terra::plot(sa_dem_rast)
+```
+
+# Calculating spectral indices
+
+Having downloaded images, we can use them to calculate spectral indices. 
+In this example, we will observe the change in vegetation between two dates for the Swarzędz area in Poland.
+Let's start by creating a new area of interest.
+
+``` {r creating swarzedz area of interest}
+swarzedz = st_point(c(17.108174, 52.405725))
+swarzedz = st_set_crs(st_sfc(swarzedz), "EPSG:4326")
+swarzedz = st_buffer(st_transform(swarzedz, "EPSG:2180"), 5000)
+```
+
+For the most accurate results, we will use Sentinel 2 data. 
+The date span is set from July to September 2023. 
+We will also pass in `composite_function = NULL`, so that all available images will be downloaded separately.
+
+``` {r downloading sentinel2 for swarzedz}
+swarzedz_sentinel2_sep = get_sentinel2_imagery(
+    swarzedz,
+    start_date = "2023-07-01",
+    end_date = "2023-09-30",
+    output_filename = tempfile(fileext = ".tif"),
+    composite_function = NULL
+)
+
+swarzedz_sentinel2_sep
+```
+
+**rsi** does not include any cloud filtering methods, but [we can use CQL2 queries](https://github.com/Permian-Global-Research/rsi/issues/6#issuecomment-1821148413) to modify **rsi**'s filter functions. 
```

Comment:
Yes, the current sentence might be misleading. I can specify that the filtering can be achieved outside of `rsi`'s scope.

Reply:
I haven't actually tested CQL2 filters yet, but... this might be possible by the end of the day :smile: 

https://github.com/Permian-Global-Research/rsi/commit/1423279b5abefa75dac7406d44cae4916161b217

Reply:
Should now be possible using the current `main` branch -- and a new vignette gives an example of using CQL2 in rsi to filter based on cloud cover :smile: 

https://permian-global-research.github.io/rsi/articles/Downloading-data-from-STAC-APIs-using-rsi.html#using-cql2-to-refine-queries-to-stac-apis